Compilation of moveit_ros/visualization fails because of boost::mutex

### Description

Compilation of moveit_ros/visualization for melodic fails because of `boost::mutex`

### Your environment
* ROS Distro: Melodic
* OS Version: Arch Linux
* Source build
* Branch melodic-devel (c2ab8a8feb6d1fca126348d5deb0bd8db1f87a8c) (but also in releases, e.g. 0.10.1, 0.9.9)

### Steps to reproduce
```bash
git clone https://github.com/ros-planning/moveit.git
cd moveit/moveit_ros/visualization
git checkout melodic-devel
mkdir build
cd build
cmake .. -DPYTHON_EXECUTABLE=/usr/bin/python3
make
```

### Expected behaviour
The package should compile successfully

### Actual behaviour
The package fails because of `boost::mutex`.


### Backtrace or Console output
```In file included from /tmp/moveit/moveit_ros/visualization/rviz_plugin_render_tools/src/trajectory_visualization.cpp:40:
/tmp/moveit/moveit_ros/visualization/rviz_plugin_render_tools/include/moveit/rviz_plugin_render_tools/trajectory_visualization.h:142:10: error: ‘mutex’ in namespace ‘boost’ does not name a type
   boost::mutex update_trajectory_message_;
          ^~~~~
/tmp/moveit/moveit_ros/visualization/rviz_plugin_render_tools/src/trajectory_visualization.cpp: In member function ‘virtual void moveit_rviz_plugin::TrajectoryVisualization::update(float, float)’:
/tmp/moveit/moveit_ros/visualization/rviz_plugin_render_tools/src/trajectory_visualization.cpp:368:12: error: ‘boost::mutex’ has not been declared
     boost::mutex::scoped_lock lock(update_trajectory_message_);
            ^~~~~
/tmp/moveit/moveit_ros/visualization/rviz_plugin_render_tools/src/trajectory_visualization.cpp: In member function ‘void moveit_rviz_plugin::TrajectoryVisualization::incomingDisplayTrajectory(const ConstPtr&)’:
/tmp/moveit/moveit_ros/visualization/rviz_plugin_render_tools/src/trajectory_visualization.cpp:491:12: error: ‘boost::mutex’ has not been declared
     boost::mutex::scoped_lock lock(update_trajectory_message_);
            ^~~~~
```

Adding `#include <boost/thread/mutex.hpp>` to `moveit_ros/visualization/rviz_plugin_render_tools/include/moveit/rviz_plugin_render_tools/trajectory_visualization.h` seams to fix the problem. But since I am not very familiar with cpp, this might not be the best solution.
Maybe this is just a problem with my OS, though. I know that Arch is not officially supported.

--- a/moveit_ros/visualization/rviz_plugin_render_tools/include/moveit/rviz_plugin_render_tools/trajectory_visualization.h
+++ b/moveit_ros/visualization/rviz_plugin_render_tools/include/moveit/rviz_plugin_render_tools/trajectory_visualization.h
@@ -37,6 +37,7 @@
 #ifndef MOVEIT_TRAJECTORY_RVIZ_PLUGIN__TRAJECTORY_VISUALIZATION
 #define MOVEIT_TRAJECTORY_RVIZ_PLUGIN__TRAJECTORY_VISUALIZATION
 
+#include <boost/thread/mutex.hpp>
 #include <moveit/macros/class_forward.h>
 #include <rviz/display.h>
 #include <rviz/panel_dock_widget.h>

